vignettes/site_only/db_connections.Rmd

```diff
@@ -0,0 +1,183 @@
+---
+title: "Database Connections using healthcare.ai"
+author: "Mike Mastanduno"
+date: "4/1/2018"
+output: html_document
+---
+
+**Database connections**
+
+Let's be honest. You need to get data from a database, do some really awesome 
+stuff with `R`, and then write the results back to a database. `R` can help
+with that. There are lots of different databases, but this document will focus
+on Microsoft SQL Server. R supports most databases, and 
+I'd encourage you to look at their [help pages](https://db.rstudio.com/) 
+for additional resources. Andiamo!
+
+If you simply want a code snippet to copy and paste, scroll to the bottom.
+
+# Reading data
+
+#### Connecting to a Database
+
+Before you can do anything, you must make a connection to your database. The 
+`build_connection_string` function takes care of the syntax for you and returns
+a connection string that's ready to be passed into `DBI::dbConnect`. After
+you have a valid connection, you should be able to browse the database using 
+RStudio's Connections pane. If you don't have a connections pane, upgrade your 
+Rstudio. It's worth it.
+
+```{r, eval=FALSE}
+library(healthcareai)
+library(DBI)
+
+my_con <- build_connection_string(server = "HCS-GM0004",
+                                  database = "SAM")
+con <- DBI::dbConnect(odbc::odbc(), .connection_string = my_con)
+```
+
+
+#### Using SQL Code
+
+From here, you can read data from the database in a couple different ways. You
+can use `db_read` to execute SQL code, or you can use `dbplyr` functionaliity
+to execute `dplyr` code directly against the database.
+```{r, eval=FALSE}
+query <- "SELECT
+          ,COUNT(*),
+          ,year(AdmitDT) as AdmitYear
+          FROM [SAM].[Encounter].[PatientEncounterBASE]
+          WHERE AdmitYear >= 2013
+          GROUP BY AdmitYear
+          ORDER BY AdmitYear"
+
+d <- db_read(con, query)
+```
+This query counts admits by year. If you don't want to pull the results into
+memory, as might the be the case with large tables, set the `pull_into_memory`
+flag to FALSE. This will create a pointer to the database that can be 
+executed "lazily." In other words R will wait until the last moment to execute 
+the statement.
+
+#### Using dplyr Code
+
+The other option is to use dplyr code. Set up a reference to the table, then use
+it to pull data. The last line, `collect()`, is what actually pulls the data 
+into memory.
+```{r, eval=FALSE}
+library(tidyverse)
+
+encounter <- tbl(con, in_schema("Encounter", "PatientEncounterBASE"))
+
+d <- encounter %>%
+  mutate(AdmitYear = year(AdmitDT)) %>%
+  filter(AdmitYear >= 2013) %>%
+  group_by(AdmitYear) %>%
+  tally() %>%
+  collect()
+```
+
+# Writing Data
+As of 4/1/2018, there are two ways to write to a database: 
```

Comment:
Lose this. Make a note in the relevant issue to edit this vignette when we have a function.

Reply:
@michaellevy People are going to ask for this as they need to do it all the time. Don't you think it's better to have example code with some caveats than nothing at all?

I think we need to have both writing sections in here for now. I don't love it, but I'd rather have the working methods and examples in a centralized place. Otherwise we're going to get people passing snippets and emails with RODBC code in them.